src/lib/device_tree/src/lib.rs

```diff
@@ -234,6 +237,76 @@ impl<'a, D: Driver> FdtIterator<'a, D> {
         Err("EOF")
     }
 }
+// Unsure about U512 as a default size. Too big?
+#[derive(Default, Debug, Clone)]
+pub struct Area {
+    pub description: String<U512>,
+    pub compatible: String<U512>,
+    // If not specified, it will be automatically computed based on previous areas (if this is
+    // first area, we start with 0).
+    pub offset: Option<u32>,
+    pub size: u32,
+    pub file: Option<String<U512>>,
+}
+
+// MAX_NAME_SIZE is 64 atm. Thus v shouldn't be able to grow beyond that.
+pub fn read_all(d: &dyn Driver) -> Vec<u8, U64> {
+    let mut v = Vec::new();
+    v.resize(MAX_NAME_SIZE, 0)
+        .expect("Tried resizing beyond v's size");
+    // Safe to unwrap because SliceReader does not return an error.
+    // as_mut_slice() is not implemented on heapless::Vec. However:
+    // "Equivalent to &mut s[..].": https://doc.rust-lang.org/std/vec/struct.Vec.html#method.as_mut_slice
+    let size = d.pread(&mut v, 0).unwrap();
+    v.truncate(size);
+    v
+}
+
+pub fn read_area_node<D: Driver>(iter: &mut FdtIterator<D>) -> Result<Area> {
```

Comment:
Could these new functions be placed in a new module src/lib/device_tree/src/area.rs since they are sort of our extension to device tree?

Reply:
Done.